yjit.c

```diff
@@ -890,6 +890,51 @@ rb_yjit_ruby2_keywords_splat_p(VALUE obj)
     return FL_TEST_RAW(last, RHASH_PASS_AS_KEYWORDS);
 }
 
+// Checks to establish preconditions for rb_yjit_splat_varg_cfunc()
+VALUE
+rb_yjit_splat_varg_checks(VALUE *sp, VALUE splat_array, rb_control_frame_t *cfp)
+{
+    if (!RB_TYPE_P(splat_array, T_ARRAY)) return Qfalse;
```

Comment:
Is this something we could optimize away if we do this using `guard_object_is_array`, or is the information often wiped out on benchmarks?

Reply:
It's optimizable, yeah. I'll toss it in.